src/lexer_tok_utils.c

```diff
@@ -56,3 +56,19 @@ t_error	ft_add_to_buf(char c, t_buf *tmp)
 	}
 	return (SUCCESS);
 }
+
+t_error	ft_add_double_redirect(t_src *src, char *c, t_buf *tmp)
+{
+	char	next;
+	t_error	err;
+
+	err = ft_peek_char(src, &next);
+	if (err != SUCCESS)
+		return (err);
+	if (ft_strchr("<>", *c) && next == *c)
```

Comment:
Since we only get into this function after checking for < and > we don't need to check the char again. So next == *c should be sufficient.

Reply:
No since we also check for \n and |